week0004/nhantd/python/week0004.py

```diff
@@ -0,0 +1,46 @@
+import sys
+
+
+class Guests():
+
+    def __init__(self):
+        self.fileRecords = dict()
+        self.printFormat = dict()
+
+
+def readFile(guests, fileName):
+    try:
+        f = open(fileName)
+        f.readline()
+        content = f.readlines()
+        for line in content:
+            firstName, lastName, email = line.strip().split(',')
+            if email in guests.fileRecords:
+                if fileName in guests.fileRecords[email]:
+                    raise Exception(
+                        'IO Error', 'Duplicated users with the same email')
+                guests.fileRecords[email].append(fileName)
+            else:
+                guests.fileRecords[email] = [fileName]
+            guests.printFormat[email] = firstName + \
+                ' ' + lastName + ' <' + email + '>'
+    except IOError:
+        raise Exception('IO Error', 'Reading file')
+
+
+def findCommonGuests(guests, firstFile, secondFile):
+
+    readFile(guests, firstFile)
+    readFile(guests, secondFile)
+    commonGuests = []
+    for key in guests.fileRecords.keys():
+        if len(guests.fileRecords[key]) == 2:
```

Comment:
Encapsulation: If you decided to use a class then you should not allow direct access to its data members, otherwise I see no need of using a class! Provide utility methods to: Add a new client, verify if registered for 2/more than 1 event, formatted string.